lib/dconf.nix

```diff
@@ -0,0 +1,15 @@
+{ lib }:
+
+# This module contains helpers for the `programs.dconf` NixOS module
+
+rec {
+  types = {
+    tuple = "_tuple";
+  };
+
+  mkTuple = _elements: {
+    inherit _elements;
+
+    _type = types.tuple;
+  };
+}
```

Comment:
Do we even need this here? Why not put it into the module.

Reply:
This incomplete implementation of GVariant type system. Ideally, we would have a full implementation like [home-manager has](https://github.com/nix-community/home-manager/blob/ec06f419af79207b33d797064dfb3fc9dbe1df4a/modules/lib/gvariant.nix) so that it can be tested.

Reply:
If that is the full one, we should keep it separate. Maybe add a comment, that it isn't complete and maybe with a ref to hm.